File: frontend/src/microservices/blog/api/LikeBlogPostPlanner.ts
// 本文件定义 LikeBlogPostPlanner，负责博客域对应接口入口。

import type { BlogPostResponse } from '@/microservices/blog/objects/BlogPostResponse'
import { executeJsonApiRequest } from '@/shared-kernel/api/ApiTransport'

export const likeBlogPost = (postId: string, payload: { userId: string }): Promise<BlogPostResponse> =>
  executeJsonApiRequest('/LikeBlogPostPlanner', 'POST', { postId, userId: payload.userId })

File: backend/src/main/scala/microservices/blog/api/LikeBlogPostPlanner.scala
// LikeBlogPostPlanner：博客域点赞博客文章入口。

package com.typesafe.travel.blog.domain

import cats.effect.IO
import com.typesafe.travel.api.routes.ConnectionApiPlan
import com.typesafe.travel.persistence.blog.BlogPlannerPlainSql

import java.sql.Connection
import java.time.Instant

object LikeBlogPostPlanner extends ConnectionApiPlan[BlogLikePlannerRequest, BlogPostResponse]:
  override val name: String = "LikeBlogPostPlanner"
  override def plan(input: BlogLikePlannerRequest, connection: Connection): IO[BlogPostResponse] =
    val now = Instant.now()
    for
      post <- BlogPlannerPlainSql.findPost(connection, input.postId, Some(input.userId))
      _ <- BlogPlannerPlainSql.likePost(connection, input.postId, input.userId, now)
      _ <-
        if post.post.authorUserId == input.userId then IO.unit
        else BlogPlannerPlainSql.insertNotification(connection, post.post.authorUserId, Some(input.userId), "postLiked", Some(input.postId), None, "Someone liked your post", now)
      updated <- BlogPlannerPlainSql.findPost(connection, input.postId, Some(input.userId))
    yield updated
